### PR TITLE
Initial implementation of VersionSpecification endpoint.

### PR DESCRIPTION
**Pull request by matthayter**
Including a `NoPackagesVersionSpecificationRepository`. Other implementations to follow.

### CODE REVIEW

File: app/com/arpnetworking/metrics/portal/version_specifications/VersionSpecificationRepository.java [hunk 1]
```diff
@@ -34,6 +34,16 @@
 public interface VersionSpecificationRepository {
```
Comment (vjkoskela) [inline app/com/arpnetworking/metrics/portal/version_specifications/VersionSpecificationRepository.java]: Not a fan of the underscore package name.

http://stackoverflow.com/questions/8243316/java-code-convention-for-package-names

I'm fine with scoping this under version or versions depending on plurality of its siblings.

File: app/com/arpnetworking/metrics/portal/version_specifications/VersionSpecificationRepository.java [hunk 2]
```diff
@@ -34,6 +34,16 @@
...
 
     /**
+     * Open the <code>AlertRepository</code>.
```
Comment (vjkoskela) [inline app/com/arpnetworking/metrics/portal/version_specifications/VersionSpecificationRepository.java]: This needs a self review pass; cut and paste error here

File: app/com/arpnetworking/metrics/portal/version_specifications/impl/NoPackagesVersionSpecificationRepository.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * @author Matthew Hayter (mhayter at groupon dot com)
```
Comment (vjkoskela) [inline app/com/arpnetworking/metrics/portal/version_specifications/impl/NoPackagesVersionSpecificationRepository.java]: Incomplete java doc. I'm ending the review here. Please polish this up and I'll have another look. If you already did a self pass consider having a peer review it as well.

Comment (matthayter) [inline app/com/arpnetworking/metrics/portal/version_specifications/impl/NoPackagesVersionSpecificationRepository.java]: Rekt. Will self-review sorry!

**Comment by vjkoskela** [comment]
@matthayter was this updated? Just checking to see if I've missed something.

**Comment by matthayter** [comment]
No updates right now, the related work has continually been pushed back.
Should be updated and completed shortly, but feel free to close in the
meantime.

**Pull request closed without merging**